src/WinRT.Runtime/Marshalers.cs

```diff
@@ -1195,13 +1197,33 @@ static Marshaler()
                     FromManaged = (T value) => value;
                     DisposeMarshaler = (object box) => { };
                     DisposeAbi = (object box) => { };
-                    CreateMarshalerArray = (T[] array) => MarshalBlittable<T>.CreateMarshalerArray(array);
-                    GetAbiArray = (object box) => MarshalBlittable<T>.GetAbiArray(box);
-                    FromAbiArray = (object box) => MarshalBlittable<T>.FromAbiArray(box);
-                    FromManagedArray = (T[] array) => MarshalBlittable<T>.FromManagedArray(array);
-                    CopyManagedArray = (T[] array, IntPtr data) => MarshalBlittable<T>.CopyManagedArray(array, data);
-                    DisposeMarshalerArray = (object box) => MarshalBlittable<T>.DisposeMarshalerArray(box);
-                    DisposeAbiArray = (object box) => MarshalBlittable<T>.DisposeAbiArray(box);
+                    if (type.IsEnum)
+                    {
+                        // For marshaling non-blittable enum arrays via MarshalNonBlittable
+                        unsafe void CopyEnum(object value, IntPtr dest) 
+                        {
+                            if (type.GetEnumUnderlyingType() == typeof(int))
+                            { 
+                                *(int*)dest.ToPointer() = (int)Convert.ChangeType(value, typeof(int));
+                            }
+                            else
+                            {
+                                *(uint*)dest.ToPointer() = (uint)Convert.ChangeType(value, typeof(uint));
+                            }
+                        }
+                        CopyAbi = (object value, IntPtr dest) => CopyEnum(value, dest);
+                        CopyManaged = (T value, IntPtr dest) => CopyEnum(value, dest);
+                    }
+                    else 
+                    { 
+                        CreateMarshalerArray = (T[] array) => MarshalBlittable<T>.CreateMarshalerArray(array);

```

Comment:
After our earlier conversation, I double checked and we seem to have a couple instances of Marshaler<T>.FromAbiArray in like IReferenceArray and IReadOnlyList and so on.  I feel like in those cases where T is an enum, we are going to end up with these functions no longer defined.  Same applies for some of the other functions.  Thoughts?

Reply:
I searched for all Marshaler<T> calls in MarshalNonBlittable and ensure that those were implemented.   So unless the calling code (MarshalNonBlittable) changes to introduce new calls, I don't see how we'd get in trouble.  Still, I'll reinstate the array functions, just in case.